src/highdicom/sr/sop.py

```diff
@@ -417,11 +419,11 @@ def __init__(
         institutional_department_name: str, optional
             Name of the department of the person or device that creates the
             SR document instance
-        verifying_observer_name: Union[str, None], optional
-            Name of the person that verfied the SR document
+        verifying_observer_name: Optional[Union[str, PersonName]], optional
```

Comment:
Thanks for spotting that `PersonName` should be fully qualified, but I don't see why we should get rid of `Optional`, as it is part of the type. Note that `Optional` and `optional` are not at all the same thing ([official docs](https://docs.python.org/3/library/typing.html#typing.Optional)) and in my opinion the docstrings should be as unambiguous as possible. 

If you prefer we could do `Union[str, pydicom.valuerep.PersonName, None]` rather than `Optional[Union[str, pydicom.valuerep.PersonName]]`? These are equivalent

@hackermd

Reply:
Now I look at it, the docstrings for many of the parameters of these methods are arguably incorrect...

Reply:
I would prefer `Union[str, pydicom.valuerep.PersonName, None], optional` over `Optional[Union[str, pydicom.valuerep.PersonName]], optional` in the docsting (I would keep `Optional` for the type annotation), since I find the double optional confusing.

In general, I think using `Union[int, float], optional` rather than `Union[int, float, None], optional` is more intuitive (while maybe not being 100% semantically correct). In the majority of cases, we want to tell users that they can either pass an `int` or `float` object to the method and `None` is an artifact of making the parameter optional.
In this particular case, it's different, because the user may actually want to set the value of the corresponding type 2 DICOM attribute to `None` (empty value).

However, I agree with you that it may be better to be unambiguous and just annotate all possible types. We may then want to revisit the docstrings throughout the library and consider using a consistent type declaration in docstrings.

Reply:
Definitely agree that it's unfortunate that "Optional[...], optional" is technically the right thing to do here, but would argue that we are sort of stuck with it. If a user is dynamically determining the parameter value using some process that may results in `None`, it is important to be unambiguous that a value of `None` can be passed, in my opinion.

Anyway, for now I have switched to `Union[str, pydicom.valuerep.PersonName, None]` in 3373a1f , as suggested.

Reply:
> Definitely agree that it's unfortunate that "Optional[...], optional" is technically the right thing to do here, but would argue that we are sort of stuck with it. If a user is dynamically determining the parameter value using some process that may results in None, it is important to be unambiguous that a value of None can be passed, in my opinion.

Agreed. Explicit is better than implicit :)